src/js/models/layerstructure.js

```diff
@@ -1035,7 +1035,7 @@ define(function (require, exports, module) {
             replace = replaceLayer &&
```

Comment:
This expression hurt my brain, can you clean it up a little?

I advise: `replace = replace || (replaceLayer && !replaceLayer.... && ...)` so we can short circuit if replace is already provided.

Also, please update the comment above so it's not stale.